### PR TITLE
add System.InhibitIdleShutdown to jsonrpc interface

### PR DESCRIPTION
**Pull request by joethefox**
This PR is an attempt to add support to the jsonrpc interface for the system command InhibitIdleShutdown. The required parameter is the "inhibit" toggle. Would be nice if @Montellese could take a look :)

### CODE REVIEW

**Comment by Tolriq** [comment]
What should the command bring ? :)

It sounds like a configuration settings ?

**Comment by joethefox** [comment]
It's a built-in function introduced with frodo, InhibitIdleShutdown(true/false) http://kodi.wiki/view/List_of_built-in_functions

File: xbmc/interfaces/json-rpc/SystemOperations.cpp [hunk 1]
```diff
@@ -95,6 +96,21 @@ JSONRPC_STATUS CSystemOperations::Reboot(const std::string &method, ITransportLa
...
 }
 
+JSONRPC_STATUS CSystemOperations::InhibitIdleShutdown(const std::string &method, ITransportLayer *transport, IClient *client, const CVariant &parameterObject, CVariant &result)
+{
+    if ((parameterObject["inhibit"].isString() &&
+         parameterObject["inhibit"].asString().compare("toggle") == 0) ||
+        (parameterObject["inhibit"].isBoolean() &&
+         parameterObject["inhibit"].asBoolean() != g_application.IsIdleShutdownInhibited()))
+    {
+        CApplicationMessenger::GetInstance().SendMsg(TMSG_INHIBITIDLESHUTDOWN, !g_application.IsIdleShutdownInhibited());
```
Comment (Montellese) [inline xbmc/interfaces/json-rpc/SystemOperations.cpp]: Too much indentation. And the curly braces aren't needed either (same in the `else if` block).

File: xbmc/interfaces/json-rpc/schema/methods.json [hunk 2]
```diff
@@ -2072,6 +2072,16 @@
     "params": [],
     "returns": "string"
   },
+  "System.InhibitIdleShutdown": {
+    "type": "method",
+    "description": "Toggle the inhibit idle shutdown timer",
```
Comment (Montellese) [inline xbmc/interfaces/json-rpc/schema/methods.json]: This description seems wrong. The functionality either inhibits or stops inhibiting the idle shutdown timer if I understood correctly.

**Comment by Montellese** [comment]
You also have to adjust the definition of `System.GetProperties` because it now supports `isidleshutdowninhibited` right? And you need to adjust the API version bump.

**Comment by Tolriq** [comment]
@Montellese since you did not answer on other ping, what is the decision for API version for krypton and later :) Early decision will make remote makers happy as they know how things will evolve.

**Comment by joethefox** [comment]
ok, I was able to mess things up :( 
I will open a new one

**Comment by Montellese** [comment]
@Tolriq: I think the best thing might be to keep the version during alphas and betas at `7.X.X` and only bump minor and patch version numbers and then jump to `8.0.0` when going to RC. If there are necessary fixes during RC we can still bump the patch version but the final release should then have a version number `8.0.X`. Does that make sense?

**Comment by Tolriq** [comment]
Well yes I guess, in the end this gives 7.x not stable 8.X stable.

And if we follow that for next case it would give the same things 9.X would be not stable then bump to 10 for stable. (So more or less Pair / Not Pair)

Important part is to know when remote makers can adapt their code to new stable API, because rewriting code 2 or 3 times if there's 3 changes during alphas would be quite a big mess :)

**Comment by Tolriq** [comment]
And @Montellese I do not know why I did say Pair all the time :p I forget to translate just this word ;) I was obviously talking about odd and even :)

**Pull request closed without merging**